task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -389,9 +392,10 @@ def wait(self) -> int:
         # If it hasn't, assume it's failed
         self._exit_code = self._exit_code if self._exit_code is not None else 1
 
-        self.client.task_instances.finish(
-            id=self.ti_id, state=self.final_state, when=datetime.now(tz=timezone.utc)
-        )
+        if self.final_state in TerminalTIState:
```

Comment:
Not 100% certain if we need this check -- I am thinking by defn `self.final_state` has to be in TerminalState -- if not it is an error!